tests/testthat/testCheckGeneSymbols.R

```diff
@@ -0,0 +1,67 @@
+# testDummy.R
+#
+# Dummy tests as a template for unit testing in test-driven development
+# for the rete project. If you test code that writes information to log-files,
+# turn logging off for all tests except for one specific block in which
+# you verify and validate your logs. Do not use your own log-file but
+# rely on the global option taht is set in the setup block. We have good
+# code samples for most test you will need - if you are not sure where to find
+# them, ask. Do not reinvent the wheel, but write your code according to the
+# principles you find in the existing files.
+#
+# Structure your tests according to the three principles expressed below:
+#    test_that("parameter errors are correctly handled", { ... })
+#    test_that("a sane input gives an expected output", { ... })
+#    test_that("a corrupt input does not lead to corrupted output", { ...})
+
+
+context("checkGeneSymbols functions")
+
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+NL <- .PlatformLineBreak()
+# ==== END SETUP AND PREPARE ===================================================
+
+test_that("isGeneSymbol identifies invalid input", {
+    # Input contains invalid input.
+    input <- c(TRUE, 3.14, data.frame(key=c(1,2)), c(1,2,3))
+    expect_error(isGeneSymbol())
+    expect_error(isGeneSymbol(c()))
+    expect_error(isGeneSymbol(NULL))
+    expect_error(isGeneSymbol(input))
+    expect_error(isGeneSymbol(TRUE))
+    expect_error(isGeneSymbol(3.14))
+    expect_error(isGeneSymbol(data.frame(key=c(1,2))))
+    expect_error(isGeneSymbol(c(1,2,3)))
+    expect_error(isGeneSymbol(NA))
```

Comment:
Hm. You an also do ```x <- as.character(NA)``` and then ```mode(x)``` is "character". Now that makes me worried that we would do so much testing here, that the fast function would be made slow again. By comparing the behaviour of match() with is.integer, I think we should do the following: if the input is NULL or length() == 0, we return FALSE:
```
if (length(input) == 0 || is.null(input) { return(FALSE) }
```
because neither of the two is a gene symbol. Otherwise we cast the input to character (which is fast, if it already character in the first place), convert to upper, and ```fmatch()``` it. Almost everything will be flattened into a vector of elements and processed. A single NA returns FALSE, and all nonsense that I could think of and tried does too.  That way we can virtually guarantee that the function returns a logical - and for example - that makes it safe to use in an ```if()``` statement. Thus the actual function would be:
```
return(!is.na(fmatch(toupper(as.character(input), geneNames$Gene_Symbol)))
```
... and if the user chooses to feed the function a bunch of numbers, that's none of our business - we'll just return a bunch of FALSE - which is correct because a number is not a gene symbol.